### PR TITLE
`qscout` compile

### PR DESCRIPTION
**Pull request by vtomole**

### CODE REVIEW

**Review by richrines1 — DISMISSED** [review]
lgtm

[for another PR] we should think about using dataclasses/subclasses of CompilerOutput  for the different endpoints

**Comment by vtomole** [comment]
@singular-value PTAL

**Comment by vtomole** [comment]
@dowusu-antwi Please review.

**Review by dowusu-antwi — DISMISSED** [review]
LGTM.

(If we don't already, we should include `notebook` as a `pip` requirement so clients / users can run these examples immediately.)

File: qiskit_superstaq/compiler_output.py [hunk 1]
```diff
@@ -74,7 +83,31 @@ def read_json(json_dict: dict, circuits_list: bool) -> AQTCompilerOutput:
...
+def read_json_qscout(json_dict: dict, circuits_list: bool) -> CompilerOutput:
+    """Reads out returned JSON from SuperstaQ API's QSCOUT compilation endpoint.
+
+    Args:
+        json_dict: a JSON dictionary matching the format returned by /qscout_compile endpoint
+        circuits_list: bool flag that controls whether the returned object has a .circuits
+            attribute (if True) or a .circuit attribute (False)
+    Returns:
+        a CompilerOutput object with the compiled circuit(s) and a list of
+        jaql programs in a string representation.
```
Comment (singular-value) [inline qiskit_superstaq/compiler_output.py:101]:
```suggestion
        jaqal programs in a string representation.
```

**Review by singular-value — APPROVED** [review]
lgtm from skim

**Comment by vtomole** [comment]
> (If we don't already, we should include notebook as a pip requirement so clients / users can run these examples immediately

We can add a `requirements.txt` in the `examples` directory that contains `notebook`; sure.

**Comment by vtomole** [comment]
@dowusu-antwi  I'll need an official approval before I can merge.

**Review by dowusu-antwi — APPROVED** [review]
LGTM.

**Pull request merged**